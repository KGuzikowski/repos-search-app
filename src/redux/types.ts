import { ThunkAction } from 'redux-thunk'
import { Action } from 'redux'

export type AppThunk<State, ReturnType = void> = ThunkAction<ReturnType, State, unknown, Action<string>>