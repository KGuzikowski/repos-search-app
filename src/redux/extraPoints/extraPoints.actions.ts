import { extraPointsActionType, acceptExtraPointsType } from './extraPoints.types'

export const acceptExtraPoints = (): acceptExtraPointsType => ({
    type: extraPointsActionType.ACCEPT_EXTRA_POINTS
})
