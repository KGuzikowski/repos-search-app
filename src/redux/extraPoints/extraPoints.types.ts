export interface extraPointsStateType {
    accepted: boolean
}

export const extraPointsActionType = {
    ACCEPT_EXTRA_POINTS: 'ACCEPT_EXTRA_POINTS'
}

export interface acceptExtraPointsType {
    type: typeof extraPointsActionType.ACCEPT_EXTRA_POINTS
}
