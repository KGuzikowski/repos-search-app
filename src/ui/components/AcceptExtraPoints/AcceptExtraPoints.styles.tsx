import styled from 'styled-components'
import { secondaryColor } from '../../style-variables'

export const AcceptPoints = styled.div`
    width: 100%;
    box-sizing: border-box;
    position: fixed;
    bottom: 0;
    padding: 20px;
    background: ${secondaryColor};
    display: flex;
    justify-content: center;
    align-items: center;
`

export const AcceptPointsMessage = styled.p`
    font-size: 20px;
    margin-right: 20px;

    @media only screen and (max-width: 500px) {
        font-size: 16px;
        padding: 10px 10px;
    }
`