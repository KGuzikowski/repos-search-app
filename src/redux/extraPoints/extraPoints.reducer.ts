import { acceptExtraPointsType, extraPointsStateType, extraPointsActionType } from './extraPoints.types'

export const INITIAL_STATE: extraPointsStateType = {
    accepted: false
}

const extraPointsReducer = (state = INITIAL_STATE, action: acceptExtraPointsType): extraPointsStateType => {
    switch (action.type) {
        case extraPointsActionType.ACCEPT_EXTRA_POINTS:
            return {
                accepted: true
            }
        default: return state
    }
}

export default extraPointsReducer