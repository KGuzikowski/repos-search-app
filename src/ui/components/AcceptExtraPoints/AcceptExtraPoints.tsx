import React from 'react'
import { useDispatch } from 'react-redux'
import { acceptExtraPoints } from '../../../redux/extraPoints/extraPoints.actions'
import { AcceptPoints, AcceptPointsMessage } from './AcceptExtraPoints.styles'
import Button from '../Button/Button'

// this component used to have a different meaning but basically it could be further developed
// to become a cookies bar so I treat that like it now.

const AcceptExtraPoints = () => {
    const dispatch = useDispatch()
    return (
        <AcceptPoints>
            <AcceptPointsMessage>
                Accept cookies!
            </AcceptPointsMessage>
            <Button reversed onClick={() => dispatch(acceptExtraPoints())}>Accept</Button>
        </AcceptPoints>
    )
}

export default AcceptExtraPoints