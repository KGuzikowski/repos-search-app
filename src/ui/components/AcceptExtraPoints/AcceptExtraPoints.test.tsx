import React from 'react'
import { create } from 'react-test-renderer'
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'
import AcceptExtraPoints from './AcceptExtraPoints'
import { extraPointsStateType } from '../../../redux/extraPoints/extraPoints.types'

it('renders AcceptExtraPoints component correctly', () => {
    const INITIAL_STATE: extraPointsStateType = {
        accepted: false
    }

    const mockStore = configureStore()
    const store = mockStore(INITIAL_STATE)

    const component = create(<Provider store={store}><AcceptExtraPoints /></Provider>)
    expect(component.toJSON()).toMatchSnapshot()
})