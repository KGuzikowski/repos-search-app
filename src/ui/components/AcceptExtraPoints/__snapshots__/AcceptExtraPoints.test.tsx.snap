// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders AcceptExtraPoints component correctly 1`] = `
<div
  className="sc-AxjAm iORRVj"
>
  <p
    className="sc-AxirZ bKXrev"
  >
    It's not cookies. Accept to add extra points!
  </p>
  <a
    className="sc-AxiKw iEFYEV"
    onClick={[Function]}
    reversed={true}
  >
    Accept
  </a>
</div>
`;
